source/dagger/gameplay/platformer/platformer_combat.cpp

```diff
@@ -0,0 +1,71 @@
+#include "platformer_combat.h"
+
+#include "core/core.h"
+#include "core/engine.h"
+#include "core/input/inputs.h"
+#include "core/graphics/sprite.h"
+#include "core/game/transforms.h"
+#include "core/graphics/animation.h"
+#include "core/graphics/shaders.h"
+#include "core/graphics/window.h"
+
+#include "gameplay/common/simple_collisions.h"
+#include "gameplay/platformer/platformer_controller.h"
+#include "gameplay/platformer/platformer_main.h"
+
+using namespace platformer;
+
+void PlatformerCombatSystem::Run()
+{
+	auto view = Engine::Registry().view <SimpleCollision, Transform, PlatformerCharacter, CharacterHealth > ();
+	for (auto entity : view) {
+		SimpleCollision& col = Engine::Registry().get<SimpleCollision>(entity);
+		Transform& t = Engine::Registry().get<Transform>(entity);
+		PlatformerCharacter& character = Engine::Registry().get<PlatformerCharacter>(entity);
+		CharacterHealth& chealth = Engine::Registry().get<CharacterHealth>(entity);
+
+		if (character.dead) {
+			return;
+		}
+		if (col.colided) {
```

Comment:
might be good to check if character in any attack state before going into collision processing

Reply:
You're right! Added checking if character has any animation active